<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><link rel="canonical" href="https://alforf.github.io/contents/The%20size%20of%20the%20octapeptidome%20space/" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>The size of the octapeptidome space - AltORF</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "The size of the octapeptidome space";
        var mkdocs_page_input_path = "contents/The size of the octapeptidome space.md";
        var mkdocs_page_url = "/contents/The%20size%20of%20the%20octapeptidome%20space/";
      </script>
    
    <script src="../../js/jquery-3.6.0.min.js" defer></script>
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
      <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> AltORF
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Contents</span></p>
              <ul class="current">
                  <li class="toctree-l1 current"><a class="reference internal current" href="./">The size of the octapeptidome space</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../The%20story%20of%20antisense%20proteins/">The story of antisense proteins</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../The%20story%20of%20cryptic%20peptides/">The story of cryptic peptides</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../The%20story%20of%20ghost%20peptides/">The story of ghost peptides</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../The%20story%20of%20microproteins/">The story of microproteins</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../The%20story%20of%20non-canonical%20protein%20expression/">The story of non canonical protein expression</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../introduction/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../table%20with%20altorf%20tools/">Table with altorf tools</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../the%20story%20of%20altorf/">The story of altorf</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../the%20story%20of%20drips/">The story of drips</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../the%20story%20of%20insDRiP/">the story of insDRiP</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../the%20story%20of%20neoantigen/">The story of neoantigen</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../the%20story%20of%20nested%20orfs/">The story of nested orfs</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../the%20story%20of%20peptons/">The story of peptons</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">AltORF</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" alt="Docs"></a> &raquo;</li>
          <li>Contents &raquo;</li>
      <li>The size of the octapeptidome space</li>
    <li class="wy-breadcrumbs-aside">
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <p>Introduction
The size of the peptidome sequence space, or the total of all possible combinations of  a peptide of N aminoacids is huge.  In order to estimate the size of the entire peptidome search space in relation to the total of potential MHCI peptides in the human genome we created a custom total translation of the human genome into octapeptides. To get a better grip on exploration of this human octapeptidome space, or all the genomic encoded octapeptides  we performed an in silico translation for the reference human genome GRCh38. Hypothetically all of these octapeptides can be transcribed and translated as the recent discoveries in the field of of non-canonical translation drastically increased the amount of potential peptides being expressed and discovered in  the MHCI peptidome. These cryptic MHC I-associated peptides (MAPs) are produced via two mechanisms: translation of protein-coding genes in non-canonical reading frames and translation of allegedly non-coding sequences.1</p>
<p>Most sequences in sequence space presumably have no function, leaving relatively small regions that are populated by naturally occurring genes. However the absence of a function may also be an interesting feature if the presumably functionless peptide encoded in the genome is expressed under upset or unusual conditions and sufficiently high enough to processed in the immune system.</p>
<p>A typical estimate of the size of entire peptidome sequence space is 20E100 (approx. 10E130) for a protein of 100 amino acids in which any of the normally occurring 20 amino acids can be found. 
MHCI is restricting our search to a defined length for peptides to 8 amino acids we will limit our calculations to peptides of 8 aa or simply 24 Nucleotides. For simplicity we also ignore post translational modifications and only assume the 20 amino acids used in normal translation. The total number of possible peptides for this octapeptidome is 20E8 = 25.600.000.000 octapeptides. This number of 25.6 billion octapeptides is referred to a the total octapeptidome. For the human genome we have determined the total number of possible peptides by using the latest update of GRCh38.p14: 3.634.521.541 unique octapeptides are generated by insilico translation of all possible reading frames and strands. For GRCh38 the number of "attempted" octapeptide translations per DNA sequence is calculated as following: </p>
<p>length of sequence - (8 codons * 3 bp/codon) + 1</p>
<p>There are 639 sequences given in GRCh38. Calculating with the above formula for each gives 3.272.074.508 theoretical octapeptide sequences population the human octapetidome. </p>
<p>Actual generated: 3.272.074.508
Unique sequences generated: 1.955.004.643
Unique "peptides" (sequences matching only 20aa + Selenocysteine): 1.241.124.045
1.241.124.045 / 3.272.074.508 ~= 0.379 ~= 37% coverage of maximum possible coverage
1.241.124.045 / 25.600.000.000 ~= 0.048 ~= 4.84% coverage of all possible sequences</p>
<p>Total peptides: 2.031.284.512
Other sequences (mismatches): 713.880.598  ?<br />
Checksum: 713.880.598 + 1.241.124.045 = 1.955.004.643</p>
<p>The most frequent peptide was KKKKKKKK with 275208. 
Here is the rest of the top 10:
247819 FFFFFFFF
226200 CVCVCVCV
214321 VCVCVCVC
209882 HTHTHTHT
209332 PPKVLGLQ
206504 THTHTHTH
200434 LPKCWDYR
192285 CNPSTLGG
186281 ASQSAGIT</p>
<p>To explain the 713.880.598 mismatches we describe the process. We open each sequence, translate the first 24bp using Biopython's translate function, record the result, then increment the position in the sequence by one. That means we also record those with stop codons or ambiguous translations. I also count them, which given the way I'm doing it also means sorting them alphabetically. After sorting and counting, we separate them by doing a search that allows only for the 21 amino acids. Here are the first 10 matches (alphabetically):</p>
<p>AAAAAAAA 4697
AAAAAAAC 37
AAAAAAAD 71
AAAAAAAE 153
AAAAAAAF 45
AAAAAAAG 521
AAAAAAAH 30
AAAAAAAI 52
AAAAAAAK 56
AAAAAAAL 169</p>
<p>If we count the number of these unique "matches" I get 1241124045. </p>
<p>All others are "mismatches". Here are the first 10 mismatches alphabetically.</p>
<p>&amp;&amp;&amp;&amp;&amp;&amp;&amp; 16802
&amp;&amp;&amp;&amp;&amp;&amp;&amp;A 133
&amp;&amp;&amp;&amp;&amp;&amp;A&amp; 6
&amp;&amp;&amp;&amp;&amp;A&amp;&amp; 1
&amp;&amp;&amp;A&amp;&amp;&amp;&amp; 1
&amp;&amp;A&amp;&amp;&amp;&amp;&amp; 6
&amp;A&amp;&amp;&amp;&amp;&amp;&amp; 8
A&amp;&amp;&amp;&amp;&amp;&amp;&amp; 48
&amp;&amp;&amp;&amp;&amp;&amp;AA 4
&amp;&amp;&amp;&amp;&amp;A&amp;A 1</p>
<p>The &amp; is the symbol for any stop codon. (If we wanted to, we could distinguish between stop codons.) So there were 16802 cases found of 8 consecutive stop codons. The total number of unique mismatches is 713.880.598.</p>
<p>Stop codons, also known as termination codons, are nucleotide triplets that signal the end of a protein-coding sequence in the messenger RNA (mRNA). The presence of eight consecutive stop codons in the human genome appears to us as a highly unusual occurrence, as the typical protein-coding gene has only one stop codon at the end of its coding sequence. It is possible that the presence of eight consecutive stop codons could be a result of sequencing errors or artifacts, rather than a genuine biological feature. </p>
<p>However, assuming that they are real, there are several potential explanations for their presence:</p>
<p>1   Regulatory element: It is possible that these stop codons are part of a regulatory element that controls gene expression. For example, they could be involved in the processing of the mRNA transcript or in the regulation of translation. 
2   Untranslated region: Another possibility is that the eight consecutive stop codons are located in the untranslated region (UTR) of a gene, rather than the coding region. The UTR is important for controlling mRNA stability and translation efficiency, and it is possible that the stop codons play a role in these processes. 
3   Non-coding RNA: The eight consecutive stop codons could also be part of a non-coding RNA molecule, rather than a protein-coding gene. Non-coding RNAs are known to play important roles in gene regulation, and it is possible that the stop codons are involved in the processing or function of the RNA molecule. 
4   Pseudogene: Finally, it is possible that the eight consecutive stop codons are located in a pseudogene, which is a non-functional copy of a gene that has accumulated mutations over time. Pseudogenes are thought to make up a significant proportion of the human genome, and some of them may contain multiple stop codons. </p>
<p>Overall, without more specific information about the location and context of the eight consecutive stop codons, it is difficult to speculate on their precise function. However, it is clear that they represent a highly unusual and potentially interesting feature of the human genome.</p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../.." class="btn btn-neutral float-left" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../The%20story%20of%20antisense%20proteins/" class="btn btn-neutral float-right" title="The story of antisense proteins">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../.." style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../The%20story%20of%20antisense%20proteins/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme_extra.js" defer></script>
    <script src="../../js/theme.js" defer></script>
      <script src="../../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
